src/server/pachw/server/pachw.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/backoff"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dlock"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/middleware/auth"
+	"github.com/pachyderm/pachyderm/v2/src/internal/miscutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/server/pfs/server"
+)
+
+const (
+	tickerSeconds      = 5
+	scaleDownThreshold = 60 / tickerSeconds
+	period             = time.Second * time.Duration(tickerSeconds)
+)
+
+type pachW struct {
+	log *logrus.Logger
+	env Env
+}
+
+func NewController(ctx context.Context, env *Env) {
+	controller := &pachW{
+		env: *env,
+		log: env.Logger,
+	}
+	go controller.run(ctx)
+}
+
+func (p *pachW) run(ctx context.Context) {
+	ctx = auth.AsInternalUser(ctx, "pachw-controller")
+	backoff.RetryUntilCancel(ctx, func() (retErr error) { //nolint:errcheck
+		lock := dlock.NewDLock(p.env.EtcdClient, path.Join(p.env.EtcdPrefix, "pachw-controller-lock"))
+		defer func() {
+			if err := lock.Unlock(ctx); err != nil {
+				unlockErr := errors.Wrap(err, "error unlocking in pachw controller")
```

Comment:
Could we use this pattern instead: https://github.com/pachyderm/pachyderm/blob/80582159e2e13ba0cb354c0d1a40fc60afd63e02/src/server/pfs/server/master.go#L131. I think the pattern of just returning errors in the callback and using the backoff for logging is a bit cleaner. This would also apply to the places below where we log as well.